### PR TITLE
Fix ICE in mir interpretation

### PR DESCRIPTION
**Pull request by oli-obk**
Indices from the end start at 1 so you can immediately subtract them from the length to get the index instead of having to do an additional `-1`. Kinda documented in https://doc.rust-lang.org/nightly/nightly-rustc/rustc/mir/enum.ProjectionElem.html#variant.ConstantIndex

### CODE REVIEW

**Comment by Centril** [comment]
cc @matthewjasper

File: src/librustc_mir/interpret/place.rs [hunk 1]
```diff
@@ -530,11 +530,12 @@ where
                     // This can only be reached in ConstProp and non-rustc-MIR.
                     throw_ub!(BoundsCheckFailed { len: min_length as u64, index: n as u64 });
                 }
-                assert!(offset < min_length);
 
                 let index = if from_end {
+                    assert!(offset - 1 < min_length);
```
Comment (RalfJung) [inline src/librustc_mir/interpret/place.rs]:
```suggestion
                    assert!(0 < offset && offset - 1 < min_length);
```
I guess?

**Pull request merged**